src/app.js

```diff
@@ -1,5 +1,198 @@
 const express = require('express');
+const strings = require('./lib/strings.js');
+const numbers = require('./lib/numbers.js');
+const bool = require('./lib/booleans.js');
+const arr = require('./lib/arrays.js');
 
 const app = express();
 
+app.use(express.json()); // FOR POST
+// app.use(bodyParser.json());
+
+app.get('/strings/hello/:basename', (req, res) => {
+  const word = strings.sayHello(req.params.basename);
+
+  res.json({ result: `${word}` }).sendStatus(200);
+});
+
+app.get('/strings/upper/:basename', (req, res) => {
+  const word = strings.uppercase(req.params.basename);
+
+  res.json({ result: `${word}` }).sendStatus(200);
+});
+
+app.get('/strings/lower/:basename', (req, res) => {
+  const word = strings.lowercase(req.params.basename);
+
+  res.json({ result: `${word}` }).sendStatus(200);
+});
+
+app.get('/strings/first-characters/:basename', (req, res) => {
+  const length = req.query.length ? parseInt(req.query.length) : 1;
+  const word = strings.firstCharacters(req.params.basename, length);
+  res.json({ result: `${word}` }).sendStatus(200);
+
+  /* if (req.query.length) {
+    const n = parseInt(Object.values(req.query).join());
+    const word = strings.firstCharacters(req.params.basename, n);
+    res.json({ result: `${word}` }).sendStatus(200);
+  } else {
+    const word = strings.firstCharacter(req.params.basename);
+    res.json({ result: `${word}` }).sendStatus(200);
+  } */
+});
+//-----------------------------------------------
+app.get('/numbers/add/:param1/and/:param2', (req, res) => {
+  const num1 = parseInt(req.params.param1);
+  const num2 = parseInt(req.params.param2);
+  const sum = numbers.add(num1, num2);
+
+  if (Number.isNaN(sum)) {
+    res.status(400).json({ error: 'Parameters must be valid numbers.' });
+  } else {
+    res.status(200).json({ result: sum });
+  }
+  // TO REMEMBER: sendStatus() closes the connection. You can only send body before it, not after. Instead use status to send body with or after status.
+});
+//--------------------------------------------------
+
+app.get('/numbers/subtract/:param1/from/:param2', (req, res) => {
+  const num1 = parseInt(req.params.param1);
+  const num2 = parseInt(req.params.param2);
+  const diff = numbers.subtract(num2, num1);
+
+  if (Number.isNaN(diff)) {
+    res.status(400).json({ error: 'Parameters must be valid numbers.' });
+  } else {
+    res.status(200).json({ result: diff });
+  }
+});
+//--------------------------------------------------
+app.post('/numbers/multiply', (req, res) => {
+  // const arr = Object.values(req.body);
+  const num1 = parseInt(req.body.a);
```

Comment:
put console.logs here

```
console.log({ num1, num2 });
```

Reply:
```
console.log src/app.js:75
      { num1: 10, num2: 3 }
    console.log src/app.js:75
      { num1: -4, num2: -9 }
    console.log src/app.js:75
      { num1: NaN, num2: NaN }
    console.log src/app.js:75
      { num1: NaN, num2: NaN }
```

Reply:
`  console.log src/app.js:75
      { num1: 10, num2: 3 }
    console.log src/app.js:75
      { num1: -4, num2: -9 }
    console.log src/app.js:75
      { num1: NaN, num2: NaN }
    console.log src/app.js:75
      { num1: NaN, num2: NaN }`

But this is multiply so it works fine anway. Shall I do the same for divide

Reply:
yes please

Reply:
```

    console.log src/app.js:110
      { div: 54 }
    console.log src/app.js:110
      { div: -0.5 }
    console.log src/app.js:110
      { div: 0 }
    console.log src/app.js:110
      { div: Infinity }
    console.log src/app.js:110
      { div: NaN }
    console.log src/app.js:110
      { div: NaN }

  ● /numbers › POST /divide › divides two 
numbers

    : Timeout - Async callback was not inv
oked within the 5000ms timeout specified b
y jest.setTimeout.Timeout - Async callback
 was not invoked within the 5000ms timeout specified by jest.setTimeout.Error:

```

Reply:
fails for 4 tests and also takes ages to run (timeout).

Reply:
I want to see a and b, not div...